colink/protocol.py

```diff
@@ -174,13 +190,18 @@ def start(self):
                         )
             queue_name = self.cl.subscribe(latest_key, start_timestamp)
             self.cl.create_entry(operator_mq_key, queue_name)
-        mq_addr = self.cl.request_info().mq_uri
-        param = pika.connection.URLParameters(url=mq_addr)
-        mq = pika.BlockingConnection(param)  # establish rabbitmq connection
-        channel = mq.channel()
-        for method, _, body in channel.consume(queue_name):
-            channel.basic_ack(method.delivery_tag)
-            data = body
+        self.cl.unlock(lock)
+        subscriber = self.cl.new_subscriber(queue_name)
+        while True:
+            try:
+                data = subscriber.get_next()
+            except redis.exceptions.ConnectionError as e:
+                if self.attached:
+                    return
```

Comment:
why directly return and exit func start? 
If redis connection error is expected, shouldn't it be something like `sleep(1s); continue`?